src/gui/guiiconprovider.h

```diff
@@ -30,30 +31,44 @@
 #ifndef GUIICONPROVIDER_H
 #define GUIICONPROVIDER_H
 
+#include "base/bittorrent/torrenthandle.h"
 #include "base/iconprovider.h"
 
 class QIcon;
 
-class GuiIconProvider : public IconProvider
+class GuiIconProvider: public IconProvider
 {
     Q_DISABLE_COPY(GuiIconProvider)
     Q_OBJECT
 
 public:
+
+    enum ThemeFlags
+    {
+        LightTheme = 1,
+        DarkTheme = 2,
+        SmallStatusIcons = 4,
+        LargeStatusIcons = 8,
+    };
+
     static void initInstance();
     static GuiIconProvider *instance();
 
+    int getThemeFlags();
     QIcon getIcon(const QString &iconId);
     QIcon getIcon(const QString &iconId, const QString &fallback);
     QIcon getFlagIcon(const QString &countryIsoCode);
-    QString getIconPath(const QString &iconId);
+    QIcon getStatusIcon(const QString &iconId);
+    QIcon getStatusIcon(const QString &iconId, BitTorrent::TorrentState state);
```

Comment:
Because the 2 functions need to call the third with different `ignoreState` flags. I could consolidate down to 2 functions, but that's allowing an implementation detail to intrude into the public API.

This is a part of the API I've struggled with. I don't want to add an "not applicable" entry to the `TorrentState` enum, since that's not actually a state. But that means that I don't have a way to say that the icon doesn't actually correspond to a state (required for the `All` filter). Hence the 2 public functions calling the private function.